src/miner.cpp

```diff
@@ -404,24 +408,38 @@ bool BlockAssembler::IsIncrementallyGood(uint64_t nExtraSize, unsigned int nExtr
         return false;
     }
 
-    // Enforce the "old" sigops for <= 1MB blocks
-    if (nBlockSize + nExtraSize <= BLOCKSTREAM_CORE_MAX_BLOCK_SIZE)
+    if (!may2020Enabled)
     {
-        // BU: be conservative about what is generated
-        if (nBlockSigOps + nExtraSigOps >= MAX_BLOCK_SIGOPS_PER_MB)
+        // Enforce the "old" sigops for <= 1MB blocks
+        if (nBlockSize + nExtraSize <= BLOCKSTREAM_CORE_MAX_BLOCK_SIZE)
         {
-            // BU: so a block that is near the sigops limit might be shorter than it could be if
-            // the high sigops tx was backed out and other tx added.
-            if (nBlockSigOps > MAX_BLOCK_SIGOPS_PER_MB - 2)
-                blockFinished = true;
-            return false;
+            // BU: be conservative about what is generated
+            if (nBlockSigOps + nExtraSigOps >= MAX_BLOCK_SIGOPS_PER_MB)
+            {
+                // BU: so a block that is near the sigops limit might be shorter than it could be if
+                // the high sigops tx was backed out and other tx added.
+                if (nBlockSigOps > MAX_BLOCK_SIGOPS_PER_MB - 2)
+                    blockFinished = true;
+                return false;
+            }
+        }
+        else
+        {
+            if (nBlockSigOps + nExtraSigOps > GetMaxBlockSigOpsCount(nBlockSize))
+            {
+                if (nBlockSigOps > GetMaxBlockSigOpsCount(nBlockSize) - 2)
+                    // very close to the limit, so the block is finished.  So a block that is near the sigops limit
+                    // might be shorter than it could be if the high sigops tx was backed out and other tx added.
+                    blockFinished = true;
+                return false;
+            }
         }
     }
-    else
+    else // may2020
```

Comment:
why we are adding this to handle the past may 2020 activation case? did we forget to do it before?